src/dao.py

```diff
@@ -0,0 +1,82 @@
+import time
+from db import session, ChatStatus, User, Chat, BagelInstance
+
+
+def print_thick_spacer():
+    print('====================')
+
+def print_thin_spacer():
+    print('--------------------')
+
+
+# User
+
+
+def user_print_all(): 
+    print_thick_spacer()
+    print('Users')
+
+    users = User.query().all()
+    for user in users:
+        print(user.name)
+
+    print_thin_spacer()
+
+
+def user_fetch(slack_id, name):
+    user = User.query().filter_by(slack_id=slack_id).first()
+    if user:
+        return user
+    else:
+        user = User(slack_id, name, True)
+        session.add(user) 
+        session.commit()
+        return user
+
+
+# Bagel Instance
+
+
+def bagel_instance_print_all():
+    print_thick_spacer()
+    print('Bagel Instances')
+
+    bagel_instances = BagelInstance.query().all()
+    for instance in bagel_instances: 
+        print('' + str(instance.id) + ' ' + str(instance.bagel_date))
```

Comment:
python >=3.6 can use `f"{str(instace.id)} {str(instacne.bagel_data)}"}` for shorthanded format strings where variables can be directly inserted in the string with curly braces `{}`

Reply:
also why the `'' + ` at the beginning?

Reply:
Oh haha the `'' +` was a java habit